### PR TITLE
hyprctl: Add shell completions

### PR DESCRIPTION
**Pull request by LOSEARDES77**
#### Describe your PR, what does it fix/add?
Add shell completions for bash and fish for the hyprctl command

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Have only tested in bash and fish shell

#### Is it ready for merging, or does it need work?
Its still WIP
Its my first time experimenting with completions so i don`t know if this is the correct way of doing it but this just works for me

Fish:
image

Bash:
image

### CODE REVIEW

**Comment by htomeht** [comment]
It would appear that this disregards nesting. It just gives the completion on the first level. 
`> hyprctl dispatch clients     │
│activewindow     getoption        output           version           │
│activeworkspace  globalshortcuts  plugin           workspacerules    │
│binds            hyprpaper        reload           workspaces        │
│clients          instances        rollinglog       -j                │
│configerrors     keyword          setcursor        -r                │
│cursorpos        kill             seterror         --batch           │
│decorations      layers           setprop          --help            │
│devices          layouts          splash           --instance        │
│dismissnotify    monitors         switchxkblayout                    │
│dispatch         notify           systeminfo `

**Comment by LOSEARDES77** [comment]
> It would appear that this disregards nesting. It just gives the completion on the first level.
> 
> > hyprctl dispatch clients
> > activewindow     getoption        output           version
> > activeworkspace  globalshortcuts  plugin           workspacerules
> > binds            hyprpaper        reload           workspaces
> > clients          instances        rollinglog       -j
> > configerrors     keyword          setcursor        -r
> > cursorpos        kill             seterror         --batch
> > decorations      layers           setprop          --help
> > devices          layouts          splash           --instance
> > dismissnotify    monitors         switchxkblayout
> > dispatch         notify           systeminfo

 i see what you mean,
I'll make it work i progress

**Comment by LOSEARDES77** [comment]
I think its ready for revision

**Comment by LOSEARDES77** [comment]
There is a issue when doing `sudo make install`   the ~ is /root, ${HOME} is /root and ${USER} is root too.
Because of this the completions files source to the root user.
I don't know how to address this

**Comment by fufexan** [comment]
Sourcing shouldn't be necessary. Shells will automatically load the completions from their default directories.

File: Makefile [hunk 1]
```diff
@@ -36,15 +36,18 @@ install:
 
 	mkdir -p ${PREFIX}/share/wayland-sessions
 	mkdir -p ${PREFIX}/bin
+	mkdir -p ${PREFIX}/share/hyprland
 	cp -f ./build/Hyprland ${PREFIX}/bin
 	cp -f ./build/hyprctl/hyprctl ${PREFIX}/bin
 	cp -f ./build/hyprpm/hyprpm ${PREFIX}/bin
+	cp -f ./hyprctl/hyprctl.bash /etc/bash_completion.d/hyprctl
```
Comment (fufexan) [inline Makefile:43]: Should be `${PREFIX}/share/...` instead of `/etc/...`

File: Makefile [hunk 2]
```diff
@@ -56,6 +59,7 @@ install:
 	cp ./subprojects/wlroots/build/libwlroots.so.13032 ${PREFIX}/lib/
 
 	$(MAKE) installheaders
+	@echo "${USER}"
```
Comment (fufexan) [inline Makefile:62]: Is this a leftover?

**Review by fufexan — APPROVED** [review]
Tested functionality in all the shells and it seems to be working fine.

I've also added the install instructions to meson.

LGTM. Thanks for the contribution!

File: hyprctl/hyprctl.usage [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            |   (-b)                                                  "Execute a batch of commands separated by ;"                   
+            ;                                                  
+
+<WINDOWS> ::= {{{ hyprctl clients | grep class | awk '{print $2}' }}}
+
+<AVIABLE_PLUGINS> ::= {{{ hyprpm list | grep "Plugin" | awk '{print $4}' }}}
+
+<MONITORS> ::= {{{ hyprctl monitors | grep Monitor | awk '{ print $2 }' }}}
+
+<KEYWORDS> ::= {{{ hyprctl devices | sed -n '/Keyboard at/{n; s/^\s\+//; p}' }}}
```
[the remark this answers is not in the document]
Comment (fufexan) [reply hyprctl/hyprctl.usage:19]: Looks like it. I'll push a fix.

**Comment by fufexan** [comment]
I was too hasty in merging this.

@LOSEARDES77 it looks like the `KEYWORDS` definition does not exist. Were you planning on adding it?

**Comment by LOSEARDES77** [comment]
it was supposed to be keyboards right
Ill change it in a moment

**Comment by LOSEARDES77** [comment]
I'll work on hyprpm completions today and add the missing keywords

**Comment by LOSEARDES77** [comment]
Can't find a way to retrieve aviable keywords

**Comment by solopasha** [comment]
https://github.com/scop/bash-completion/blob/master/README.md
bash completions should be installed in
```
❯ pkg-config --variable=completionsdir bash-completion
/usr/share/bash-completion/completions
```

for fish:
from https://fishshell.com/docs/current/completions.html:
>A directory for third-party software vendors to ship their own completions for their software, usually /usr/share/fish/vendor_completions.d;

**Comment by LOSEARDES77** [comment]
I see ill applt suggested changes to #5423

**Comment by Ordoviz** [comment]
This could be improved a little further by taking inspiration from #3535

**Comment by fufexan** [comment]
@Ordoviz I completely missed that PR, apologies. Can you rebase it and make whatever changes you see fit?

**Pull request merged**